meteor/client/ui/Settings/ShowStyle/VariantListItem.tsx

```diff
@@ -36,6 +37,15 @@ export const VariantListItem: React.FunctionComponent<IShowStyleVariantItemProps
 	const initialEditedMappings: ProtectedString<'ShowStyleVariantId'>[] = []
 	const [editedMappings, setEditedMappings] = useState(initialEditedMappings)
 
+	const showStyleVariantConfigurationVerifier: ShowStyleVariantConfigurationVerifier =
+		new ShowStyleVariantConfigurationVerifier()
+	const isBlueprintConfigurationInvalid: boolean =
+		showStyleVariantConfigurationVerifier.isBlueprintConfigurationSelectedFromBaseInvalid(
+			props.showStyleVariant,
+			props.showStyleBase,
+			props.blueprintConfigManifest
+		)
```

Comment:
You could set it up as a static singleton on the class, such that it can be accessed as `ShowStyleVariantConfigurationVerifier.instance`.

Reply:
I've updated it so that `ShowStyleVariantConfigurationVerifier` is now exporting an instance so it should only be created once now.

The ideal solution would still be to add Dependency Injection to Sofie, but that is a task of its own.